### DISCUSSION TITLE
Welcome to Onekey Discussions!

### DISCUSSION THREAD

**Discussion by ikun0014** [ikunshare/Onekey#26] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 10086pro** [comment]
是否考虑添加辅助入库的JavaScript？
// ==UserScript==
// @name         辅助入库
// @namespace    http://tampermonkey.net/
// @version      1.0
// @description  在特定网站添加辅助入库功能。
// @match        *://steamdb.info/app/*
// @match        *://store.steampowered.com/app/*
// @match        *://steamui.com/*
// @match        *://*.steamui.com/*
// @match        *://*.g2a.com/*
// @match        *://*.humblebundle.com/*
// @match        *://*.greenmangaming.com/*
// @match        *://*.instant-gaming.com/*
// @match        *://steamcharts.com/*
// @match        *://steamstat.us/*
// @run-at       document-end
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    // 创建按钮元素，指定文本和appId
    function createButton(appId, buttonText) {
        const button = document.createElement('button');
        button.textContent = buttonText;
        button.className = 'enhanced-button';
        button.onclick = function() {
            handleButtonClick(appId);
        };
        return button;
    }

    // 处理按钮点击事件
    function handleButtonClick(appId) {
        console.log(`辅助入库功能正在开发中，appId: ${appId}`);
        // 实际的辅助入库逻辑将在这里实现
    }

    // 安全地向特定元素添加按钮
    function safeAddButton(parent, button) {
        try {
            if (parent) {
                parent.appendChild(button);
            } else {
                console.error('父元素未找到，无法添加按钮');
            }
        } catch (error) {
            console.error('添加按钮失败:', error);
        }
    }

    // 根据页面类型添加辅助入库按钮
    function addEnhancedButtons() {
        const appId = getAppIdFromUrl();
        if (!appId) {
            console.warn('在URL中未找到AppId。');
            return;
        }

        // 检查页面上是否存在可以添加按钮的位置，并添加按钮
        const possibleParents = document.querySelectorAll('header, footer, .game-header, .game-info, .apphub_OtherSiteInfo, .game-item');
        possibleParents.forEach(parent => {
            if (parent) {
                const existingButton = parent.querySelector('.enhanced-button');
                if (!existingButton) {
                    const button = createButton(appId, '辅助入库');
                    safeAddButton(parent, button);
                }
            }
        });
    }

    // 从当前URL中提取appId
    function getAppIdFromUrl() {
        const url = window.location.href;
        const match = url.match(/\/app\/(\d+)|\/(\d+)-[^\/]*\/?$/);
        return match ? parseInt(match[1] || match[2], 10) : null;
    }

    // 观察页面变化并添加按钮
    function observeAndAddButtons() {
        const observer = new MutationObserver((mutations) => {
            mutations.forEach(mutation => {
                if (mutation.addedNodes.length) {
                    addEnhancedButtons();
                }
            });
        });
        observer.observe(document.body, {
            childList: true,
            subtree: true
        });
    }

    // 主执行开始
    addEnhancedButtons();
    observeAndAddButtons();
})();
满城百姓，感谢不尽！

**Reply by ikun0014** [reply]
好的，下个版本将尝试支持

**Reply by qingchengluoxue** [reply]
能不能出个彻底删除卸载游戏残留文件的工具，谢谢大佬
